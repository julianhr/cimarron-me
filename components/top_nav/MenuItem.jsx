/* @jsx jsx */
import React from 'react'
import PropTypes from 'prop-types'
import Link from 'next/link'
import { jsx } from '@emotion/core'
import styled from '@emotion/styled'

import { colors } from '../../library'


const Root = styled.div`
  height: 100%;
  display: flex;
  align-items: center;
`

const styles = {
  link: {
    color: `${colors.TEXT}`,
    textDecoration: 'none',
    margin: '0 10px',
    padding: '0 6px 4px',
    borderBottomWidth: 2,
    borderBottomStyle: 'solid',
    borderBottomColor: 'transparent',
    ':hover': {
      borderBottomColor: '#e10d75',
      transitionDuration: '140ms',
    },
    transitionProperty: 'border-bottom-color',
    transitionDuration: '75ms',
    transitionTimingFunction: 'linear',
  }
}

function MenuItem(props) {
  return (
    <Root>
      <Link
        href={props.url}
      >
        <a css={styles.link}>{props.label}</a>
      </Link>
    </Root>
  )
}

MenuItem.propTypes = {
  url: PropTypes.string,
  label: PropTypes.string,
}

export default MenuItem
