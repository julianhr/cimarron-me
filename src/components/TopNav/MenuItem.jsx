import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import styled from '@emotion/styled'


const Root = styled.div`
  height: 100%;
  display: flex;
  align-items: center;
`

const StyledLink = styled(Link)`
    color: ${props => props.theme.colors.text};
    text-decoration: none;
    margin: 0 10px;
    padding: 0 6px 4px;
    border-bottom: 2px solid transparent;
    transition: border-bottom-color 75ms linear;

    :hover {
      border-bottom-color: #e10d75;
      transition-duration: 140ms;
    }
`

function MenuItem(props) {
  return (
    <Root>
      <StyledLink
        to={props.url}
      >
        {props.label}
      </StyledLink>
    </Root>
  )
}

MenuItem.propTypes = {
  url: PropTypes.string,
  label: PropTypes.string,
}

export default MenuItem
