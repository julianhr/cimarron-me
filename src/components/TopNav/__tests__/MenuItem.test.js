import React from 'react'
import { render, mount, shallow } from 'enzyme'
import { create } from 'react-test-renderer'

import MenuItem from '../MenuItem'
import MockApp from '~/__tests__/__mocks__/MockApp'


const testProps = {
  url: '/some/test/path',
  label: 'test label',
  isLinkRouted: false,
}

const renderApp = props => create(<MockApp><MenuItem {...props} /></MockApp>)

describe('MenuItem', () => {
  describe('props', () => {
    it('requires url', () => {
      const props = {...testProps}
      delete props.url
      expect(() => renderApp(props)).toThrow()
    })

    it('requires label', () => {
      const props = {...testProps}
      delete props.label
      expect(() => renderApp(props)).toThrow()
    })

    it('isLinkRouted is not required', () => {
      const props = {...testProps}
      delete props.isLinkRouted
      expect(() => renderApp(props)).not.toThrow()
    })
  })

  describe('snapshots', () => {
    it('isLinkRouted == false', () => {
      const props = {...testProps}
      props.isLinkRouted = false
      const wrapper = mount(
        <MockApp>
          <MenuItem {...props} />
        </MockApp>
      )
      expect(wrapper).toMatchSnapshot()
      wrapper.unmount()
    })

    it('isLinkRouted == true', () => {
      const props = {...testProps}
      props.isLinkRouted = true
      const wrapper = mount(
        <MockApp>
          <MenuItem {...props} />
        </MockApp>
      )
      expect(wrapper).toMatchSnapshot()
      wrapper.unmount()
    })
  })
})
