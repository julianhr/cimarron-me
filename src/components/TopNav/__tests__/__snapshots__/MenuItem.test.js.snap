// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem snapshots isLinkRouted == false 1`] = `
<Root>
  <div
    className="css-u96gbo-Root e1uo4w6s0"
  >
    <EmotionCssPropInternal
      __EMOTION_LABEL_PLEASE_DO_NOT_USE__="AnchorTag"
      __EMOTION_TYPE_PLEASE_DO_NOT_USE__="a"
      css={[Function]}
      href="/some/test/path"
      rel="noopener noreferrer"
      target={null}
    >
      <a
        className="css-d3pkg8-STYLES-STYLES"
        href="/some/test/path"
        rel="noopener noreferrer"
        target={null}
      >
        test label
      </a>
    </EmotionCssPropInternal>
  </div>
</Root>
`;

exports[`MenuItem snapshots isLinkRouted == true 1`] = `
<Root>
  <div
    className="css-u96gbo-Root e1uo4w6s0"
  >
    <EmotionCssPropInternal
      __EMOTION_LABEL_PLEASE_DO_NOT_USE__="AnchorTag"
      __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
      css={[Function]}
      to="/some/test/path"
    >
      <a
        className="css-d3pkg8-STYLES-STYLES"
        href="/some/test/path"
        onClick={[Function]}
      >
        test label
      </a>
    </EmotionCssPropInternal>
  </div>
</Root>
`;
